From 8ee0a75a1f7ae044cd732512730ba64eddbf6714 Mon Sep 17 00:00:00 2001
From: Felix Kuehling <Felix.Kuehling@amd.com>
Date: Thu, 12 May 2016 13:27:52 -0400
Subject: [PATCH 1556/1722] drm/amdgpu: Fix extra BO reference causing memory
 leak

amdgpu_bo_create creates the BO with a reference. Don't take another
unbalanced reference before returning from amdgpu_gem_object_create.

Change-Id: I25837adfc40536da47d0f3e6995892a87d19f3a3
Signed-off-by: Felix Kuehling <Felix.Kuehling@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_gem.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_gem.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_gem.c
index 75cdb03..5d5099f 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_gem.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_gem.c
@@ -105,7 +105,7 @@ retry:
 	}
 
 	list_add(&gobj->list, &robj->gem_objects);
-	gobj->bo = amdgpu_bo_ref(robj);
+	gobj->bo = robj;
 	*obj = &gobj->base;
 
 
-- 
2.7.4

